File: python/device/arduino-yun/examples/coap_observe.py
import time
import os
import random
import json
import signal
import sys
from lib import qiot
"""
	This sample code demo receive value from QIoT Suite Lite by CoAP protocol
	requirement:
	-- opkg update
	-- opkg install distribute
	-- opkg install python-openssl
	-- easy_install pip
	-- pip install coapthon
	run command: python coap_observe.py
"""

"""
	Setup connection options
"""
connection = qiot.connection(qiot.protocol.COAP)
connection_options = connection.read_resource('./res/resourceinfo.json')

"""
	Receive data of QIoT Suite Lite.
"""
def on_message(event_trigger,data):
	message =json.loads(data["message"])
	if data['id']=='temp':
		print "temp : " + str(message['value'])
		print "------------------------"

connection.on("message",on_message)
connection.connect(connection_options)
connection.subscribe_by_id("temp")

def signal_handler(signal, frame):
        print('You pressed Ctrl+C!')
        connection.stop()
        
signal.signal(signal.SIGINT, signal_handler)

while 1:
	time.sleep(1)

File: nodejs/device/intel-edison/examples/coap-observe.js
/**
 *   This sample code demo receive value from QIoT Suite Lite by CoAP protocol
 *   requirement:
 *   -- npm install
 *   run command: node coap-observe.js
 */

var qiot = require('./lib/qiot');

/**
 * Setup connection options
 */
var connection = new qiot(qiot.protocol.COAP);
var connection_option = connection.readResource('./res/resourceinfo.json');

function getRandomInt(min, max) {
    return Math.floor(Math.random() * (max - min + 1)) + min;
}

/** 
 * Receive data of QIoT Suite Lite.
 */
connection.on('message', function(data) {
    switch (data.id) {
        case "temp":
            console.log("------------------------")
            console.log("temp : " + data.message.value);
            console.log("------------------------")
            break;
        default:
            break;
    }
});

/**
 * Subscribe data coming from QIoT Suite Lite.
 */
// TODO:you could replace "temp" by any resource id set form QIoT Suite Lite
connection.subscribeById("temp");